File: sdk/coinrailzClient.py
"""Coin Railz x402 SDK - Python Client"""
import json
import base64
import requests
from typing import Dict, List, Any

COINRAILZ_BASE_URL = "https://coinrailz.com/api/x402"
PLATFORM_WALLET = "0xa4bbe37f9a6ae2dc36a607b91eb148c0ae163c91"
USDC_BASE = "0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913"
BASE_CHAIN_ID = 8453

def encode_x_payment(tx_hash: str) -> str:
    """Encode payment proof for x402 protocol"""
    payload = {"txHash": tx_hash}
    return base64.b64encode(json.dumps(payload).encode("utf-8")).decode("utf-8")

def wallet_risk(wallet_address: str, chain: str, tx_hash: str) -> Dict[str, Any]:
    """Check wallet risk score"""
    x_payment = encode_x_payment(tx_hash)
    res = requests.post(
        f"{COINRAILZ_BASE_URL}/wallet-risk",
        headers={"Content-Type": "application/json", "X-PAYMENT": x_payment},
        json={"walletAddress": wallet_address, "chain": chain}
    )
    res.raise_for_status()
    return res.json()

class CoinRailzClient:
    """Main client for Coin Railz x402 services"""
    
    def __init__(self, base_url: str = COINRAILZ_BASE_URL):
        self.base_url = base_url
        self.platform_wallet = PLATFORM_WALLET

    def wallet_risk(self, wallet_address: str, chain: str, tx_hash: str) -> Dict[str, Any]:
        return wallet_risk(wallet_address, chain, tx_hash)

    def gas_price_oracle(self, chains: List[str], tx_hash: str) -> Dict[str, Any]:
        x_payment = encode_x_payment(tx_hash)
        res = requests.post(
            f"{self.base_url}/gas-price-oracle",
            headers={"Content-Type": "application/json", "X-PAYMENT": x_payment},
            json={"chains": chains}
        )
        res.raise_for_status()
        return res.json()

File: sdk/coinrailzClient.ts
// Coin Railz x402 SDK - TypeScript Client
import fetch from "node-fetch";
import { Buffer } from "buffer";

const COINRAILZ_BASE_URL = "https://coinrailz.com/api/x402";
const PLATFORM_WALLET = "0xa4bbe37f9a6ae2dc36a607b91eb148c0ae163c91";
const USDC_BASE = "0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913";
const BASE_CHAIN_ID = 8453;

export type Chain = "base";

export interface WalletRiskParams {
  walletAddress: string;
  chain: Chain;
}

export interface CoinRailzPaymentPayload {
  txHash: string;
}

export function encodeXPayment(payload: CoinRailzPaymentPayload): string {
  return Buffer.from(JSON.stringify(payload)).toString("base64");
}

export async function callWalletRisk(params: WalletRiskParams, txHash: string) {
  const xPayment = encodeXPayment({ txHash });

  const res = await fetch(`${COINRAILZ_BASE_URL}/wallet-risk`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "X-PAYMENT": xPayment
    },
    body: JSON.stringify(params)
  });

  if (!res.ok) {
    const text = await res.text();
    throw new Error(`Coin Railz wallet-risk error (${res.status}): ${text}`);
  }

  return res.json();
}

export class CoinRailzClient {
  private baseUrl: string;
  private platformWallet: string;

  constructor(baseUrl?: string) {
    this.baseUrl = baseUrl || COINRAILZ_BASE_URL;
    this.platformWallet = PLATFORM_WALLET;
  }

  async walletRisk(params: WalletRiskParams, txHash: string) {
    return callWalletRisk(params, txHash);
  }

  async gasPriceOracle(chains: string[], txHash: string) {
    const xPayment = encodeXPayment({ txHash });
    const res = await fetch(`${this.baseUrl}/gas-price-oracle`, {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        "X-PAYMENT": xPayment
      },
      body: JSON.stringify({ chains })
    });

    if (!res.ok) {
      const text = await res.text();
      throw new Error(`Coin Railz gas-price-oracle error (${res.status}): ${text}`);
    }

    return res.json();
  }
}
